homeassistant/components/spider/__init__.py

```diff
@@ -1,57 +1,94 @@
 """Support for Spider Smart devices."""
-from datetime import timedelta
+import asyncio
 import logging
 
 from spiderpy.spiderapi import SpiderApi, UnauthorizedException
 import voluptuous as vol
 
+from homeassistant.config_entries import SOURCE_IMPORT
 from homeassistant.const import CONF_PASSWORD, CONF_SCAN_INTERVAL, CONF_USERNAME
 import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.discovery import load_platform
 
-_LOGGER = logging.getLogger(__name__)
-
-DOMAIN = "spider"
-
-SPIDER_COMPONENTS = ["climate", "switch"]
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN, PLATFORMS
 
-SCAN_INTERVAL = timedelta(seconds=120)
+_LOGGER = logging.getLogger(__name__)
 
 CONFIG_SCHEMA = vol.Schema(
     {
         DOMAIN: vol.Schema(
             {
                 vol.Required(CONF_PASSWORD): cv.string,
                 vol.Required(CONF_USERNAME): cv.string,
-                vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL): cv.time_period,
+                vol.Optional(
+                    CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+                ): cv.time_period,
             }
         )
     },
     extra=vol.ALLOW_EXTRA,
 )
 
 
-def setup(hass, config):
-    """Set up Spider Component."""
+def _spider_startup_wrapper(entry):
+    """Startup wrapper for spider."""
+    api = SpiderApi(
+        entry.data[CONF_USERNAME],
+        entry.data[CONF_PASSWORD],
+        entry.data[CONF_SCAN_INTERVAL],
+    )
+    return api
 
-    username = config[DOMAIN][CONF_USERNAME]
-    password = config[DOMAIN][CONF_PASSWORD]
-    refresh_rate = config[DOMAIN][CONF_SCAN_INTERVAL]
 
-    try:
-        api = SpiderApi(username, password, refresh_rate.total_seconds())
+async def async_setup(hass, config):
+    """Set up a config entry."""
+    hass.data[DOMAIN] = {}
+    if DOMAIN not in config:
+        return True
+
+    conf = config.get(DOMAIN, {})
```

Comment:
We don't need to use `dict.get` if we know that the key is in the dict. Please use `dict[key]`.